internal/cmd/config/config.go

```diff
@@ -1278,6 +1285,16 @@ func (c *Config) SetupControllerPublicClusterAddress(flagValue string) error {
 			return fmt.Errorf("Error splitting public cluster adddress host/port: %w", err)
 		}
 	}
+
+	// remove the square brackets from the ipv6 address because the method
+	// net.JoinHostPort() will add a second pair of square brackets.
+	host = squareBrackets.ReplaceAllString(host, "")
+	if host == "" {
+		// return early if a empty host was provided
+		c.Controller.PublicClusterAddr = net.JoinHostPort(host, port)
+		return nil
+	}
+
 	c.Controller.PublicClusterAddr = net.JoinHostPort(host, port)
 	return nil
```

Comment:
I am curious what the purpose is of this return early clause if we return immediately afterwards

Reply:
Thanks for catching this. My original implementation parsed the host value to check if it was a valid IPv4 or IPv6 address, but I had to remove this logic because it would break some unix listener and enos tests. This if conditional was an edge case to ensure that these tests passed as expected, but now I can remove this logic.